universal-application-tool-0.0.1/app/services/applicant/question/QuestionImpl.java

```diff
@@ -0,0 +1,28 @@
+package services.applicant.question;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableSet;
+import services.question.types.QuestionType;
+
+abstract class QuestionImpl implements Question {
+  protected final ApplicantQuestion applicantQuestion;
+
+  public QuestionImpl(ApplicantQuestion applicantQuestion) {
+    this.applicantQuestion = applicantQuestion;
+    if (!validQuestionTypes().contains(applicantQuestion.getType())) {
+      throw new RuntimeException(
+          String.format(
+              "Question is not a question of the following types: [%s]: %s (type: %s)",
+              Joiner.on(", ").join(validQuestionTypes().stream().toArray()),
+              applicantQuestion.getQuestionDefinition().getQuestionPathSegment(),
+              applicantQuestion.getQuestionDefinition().getQuestionType()));
+    }
+  }
+
+  protected abstract ImmutableSet<QuestionType> validQuestionTypes();
+
+  @Override
+  public boolean isAnswered() {
```

Comment:
Added. I believe I addressed the skip comment you mentioned, but please let me know if I didn't quite address your concern.

Reply:
aha, the second sentence isn't quite there yet since it's the applicant data that determines whether the question is answered, skipped, or unanswered. maybe replace the second sentence with "If the applicant data does not contain the question's path, then i twill be considered unanswered."

Reply:
I like that suggestion and have incorporated it.